src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -123,6 +123,13 @@ class OverlayImpl : public Overlay
     std::atomic <uint64_t> peerDisconnects_ {0};
     std::atomic <uint64_t> peerDisconnectsCharges_ {0};
 
+    // Last time we crawled peers for shard info
+    std::atomic<std::chrono::seconds> csLast_{std::chrono::seconds{0}};
```

Comment:
`std::atomic` doesn't play well time_point. I can't remember the reason why. It may have been that atomic requires trivially copyable types, and time_point isn't trivially copyable.

Reply:
Paging @HowardHinnant.

Reply:
I remember this happening, and I don't recall the reason either.  One can always load the duration into a time_point, do the computations, and then extract the duration from the time_point for storing.

Reply:
@HowardHinnant Yeah, that is the solution I went with.